dart/lib/src/scope.dart

```diff
@@ -213,6 +213,26 @@ class Scope {
       extra.map((key, value) => MapEntry(key, value))
         ..addAll(event.extra ?? {});
 
+  /// If scope and event have a user, the user of the event takes
+  /// precedence.
+  SentryUser? _mergeUsers(SentryUser? scopeUser, SentryUser? eventUser) {
+    if (scopeUser == null && eventUser != null) {
+      return eventUser;
+    }
+    if (eventUser == null && scopeUser != null) {
+      return scopeUser;
+    }
+    // otherwise the user of scope takes precedence over the event user
+    return scopeUser?.copyWith(
+      id: eventUser?.id,
+      email: eventUser?.email,
+      ipAddress: eventUser?.ipAddress,
+      username: eventUser?.username,
+      // TODO should extras be merged?
```

Comment:
I'm not quite sure if the extras of the user should be merged

Reply:
I guess we can do, yes, but event extras keys have precedence